pkg/endpoint/endpoint.go

```diff
@@ -2403,3 +2461,63 @@ func (e *Endpoint) IPs() []net.IP {
 func (e *Endpoint) InsertEvent() {
 	e.getLogger().Info("New endpoint")
 }
+
+// syncPolicyMap attempts to synchronize the PolicyMap for this endpoint to
+// contain the set of PolicyKeys represented by the endpoint's desiredMapState.
+// It checks the current contents of the endpoint's PolicyMap and deletes any
+// PolicyKeys that are not present in the endpoint's desiredMapState. It then
+// adds any keys that are not present in the map. When a key from desiredMapState
+// is inserted successfully to the endpoint's BPF PolicyMap, it is added to the
+// endpoint's realizedMapState field. Returns an error if the endpoint's BPF
+// PolicyMap is unable to be dumped, or any update operation to the map fails.
+// Must be called with e.Mutex locked.
+func (e *Endpoint) syncPolicyMap() error {
+
+	if e.realizedMapState == nil {
+		e.realizedMapState = make(map[policymap.PolicyKey]struct{})
+	}
+
+	currentMapContents, err := e.PolicyMap.DumpToSlice()
+
+	if err != nil {
+		return fmt.Errorf("unable to dump PolicyMap for endpoint: %s", err)
+	}
+
+	for _, entry := range currentMapContents {
+
+		// If key that is in policy map is not in desired state, just remove it.
+		if _, ok := e.desiredMapState[entry.Key]; !ok {
+			err := e.PolicyMap.DeleteKey(entry.Key)
+			if err != nil {
+				return err
+			}
+			// Remove from realized state
+			delete(e.realizedMapState, entry.Key)
+		}
+	}
+
+	for keyToAdd := range e.desiredMapState {
+		if _, ok := e.realizedMapState[keyToAdd]; !ok {
+			err := e.PolicyMap.AllowKey(keyToAdd)
+			if err != nil {
+				return err
+			}
+			e.realizedMapState[keyToAdd] = struct{}{}
+		}
+	}
+	return nil
+}
+
+func (e *Endpoint) syncPolicyMapController() {
+	ctrlName := fmt.Sprintf("sync-policymap-%d", e.ID)
+	e.controllers.UpdateController(ctrlName,
+		controller.ControllerParams{
+			DoFunc: func() error {
+				e.Mutex.Lock()
+				defer e.Mutex.Unlock()
+				return e.syncPolicyMap()
+			},
+			RunInterval: time.Duration(1) * time.Minute,
```

Comment:
No need to convert constants into `time.Duration`.
This would be more readable:
```
RunInterval: 1 * time.Minute,
```